http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -1303,8 +1303,8 @@ static void recordRequestComplete(RequestStats requestStats, Request request, lo
             return;
         }
 
-        Duration responseProcessingTime = Duration.nanosSince(responseStart);
-        Duration requestProcessingTime = new Duration(responseStart - requestStart, NANOSECONDS);
+        Duration responseProcessingTime = new Duration(max(0, System.nanoTime() - responseStart), NANOSECONDS);
```

Comment:
Have you seen any cases where it's negative? If duration is obtained from System.nanoTime(), the difference should never be negative.

Reply:
@martint yes we did

Reply:
```
Caused by: java.lang.IllegalArgumentException: value is negative: -243459.0
	at io.airlift.units.Duration.<init>(Duration.java:79)
	at io.airlift.units.Duration.succinctDuration(Duration.java:66)
	at io.airlift.units.Duration.succinctNanos(Duration.java:58)
	at io.airlift.units.Duration.nanosSince(Duration.java:53)
	at io.airlift.http.client.jetty.JettyHttpClient.recordRequestComplete(JettyHttpClient.java:1306)
	at io.airlift.http.client.jetty.JettyHttpClient.lambda$buildCompletionHandler$12(JettyHttpClient.java:880)
```

Reply:
@martint the jetty code is super simple to analyze by looking at it, it operates on delta between two System.nanoTime()-s. I can't find anything wrong there

Reply:
@martint but it is